tf2_geometry_msgs/test/test_tf2_geometry_msgs.cpp

```diff
@@ -520,22 +526,20 @@ TEST(TfGeometry, Quaternion)
     q1.header.frame_id = "A";
 
     // simple api
-    const geometry_msgs::msg::QuaternionStamped q_simple = tf_buffer->transform(
-      q1, "B", tf2::durationFromSec(
-        2.0));
-    EXPECT_NEAR(q_simple.quaternion.x, M_SQRT1_2, EPS);
-    EXPECT_NEAR(q_simple.quaternion.y, 0, EPS);
-    EXPECT_NEAR(q_simple.quaternion.z, -1 * M_SQRT1_2, EPS);
-    EXPECT_NEAR(q_simple.quaternion.w, 0, EPS);
+    {
+      const geometry_msgs::msg::QuaternionStamped q_simple = tf_buffer->transform(
```

Comment:
It might help the reviewers, and readability, if you articulate what's going on in this function call. I've been looking at this for ten minutes and it still isn't quite clear what's going on.

Reply:
I added an intermediate variable - I think the diff moved your comment though, which line specifically is confusing?

Reply:
Eh, that doesn't really tell me what the test does, it just tells me what the result should be. What would really help readers is something like, `// Frame a and frame b differ by [x,y,z] and [p,q,r] at time t, check the resulting quat is <foo>` 

That's why I dislike this big ball of wax test pattern, when you write one function per test you can name the tests something sane like, "check_basic_frame_transform" and it all makes sense. The original author really should have at least left us a few bread crumb comments. 😢 

Anyway, not your fault, thanks for putting up with this stuff.